cpp/oneapi/dal/io/csv/detail/read_ops.hpp

```diff
@@ -17,36 +17,85 @@
 #pragma once
 
 #include "oneapi/dal/table/common.hpp"
+#include "oneapi/dal/graph/common.hpp"
 #include "oneapi/dal/io/csv/read_types.hpp"
+#include "oneapi/dal/io/csv/detail/select_kernel.hpp"
 
 namespace oneapi::dal::csv::detail {
 namespace v1 {
 
 template <typename Object, typename Policy, typename... Options>
-struct read_ops_dispatcher {
-    Object operator()(const Policy&, const data_source_base&, const read_args<Object>&) const;
+struct read_ops_dispatcher;
+
+template <typename Object>
+struct read_ops_dispatcher<Object, dal::detail::host_policy> {
+    Object operator()(const dal::detail::host_policy& policy,
+                      const data_source_base& ds,
+                      const dal::preview::csv::read_args<Object>& args) const {
+        static auto impl =
+            dal::preview::csv::detail::get_backend<dal::detail::host_policy,
+                                                   data_source_base,
+                                                   dal::preview::csv::read_args<Object>>(ds, args);
+        return (*impl)(policy, ds, args);
+    }
 };
 
+template <>
+struct read_ops_dispatcher<table, dal::detail::host_policy> {
+    table operator()(const dal::detail::host_policy& policy,
+                     const data_source_base& ds,
+                     const dal::csv::read_args<table>& args) const;
+};
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+template <>
+struct read_ops_dispatcher<table, dal::detail::data_parallel_policy> {
+    table operator()(const dal::detail::data_parallel_policy& ctx,
+                     const data_source_base& ds,
+                     const dal::csv::read_args<table>& args) const;
+};
+
+#endif
```

Comment:
TIP: These can be merged into one partially specialized template:
```c++
template <typename Policy>
struct read_ops_dispatcher<table, Policy> {
    table operator()(const Policy& policy,
                     const data_source_base& ds,
                     const dal::csv::read_args<table>& args) const;
};
```

Reply:
In this case we will have two partial specializations
```
template <typename Object>
struct read_ops_dispatcher<Object, dal::detail::host_policy> {
```
and
```
template <typename Policy>
struct read_ops_dispatcher<table, Policy> {
```
it causes the ambiguous template instantiation for `struct read_ops_dispatcher<table, host_policy>`